src/backend/renderer/utils.rs

```diff
@@ -471,200 +488,3 @@ where
     );
     result
 }
-
-#[derive(Debug, Default)]
-struct RenderOp {
-    src: Rectangle<f64, BufferCoord>,
-    dst: Rectangle<i32, Physical>,
-    damage: Vec<Rectangle<i32, Physical>>,
-}
-
-/// Draws a surface and its subsurfaces using a given [`Renderer`] and [`Frame`].
-///
-/// - `scale` needs to be equivalent to the fractional scale the rendered result should have.
-/// - `location` is the position the surface should be drawn at.
-/// - `damage` is the set of regions that should be drawn relative to the same origin as the location.
-///
-/// Note: This element will render nothing, if you are not using
-/// [`crate::backend::renderer::utils::on_commit_buffer_handler`]
-/// to let smithay handle buffer management.
-#[allow(clippy::too_many_arguments)]
-pub fn draw_surface_tree<R, S>(
```

Comment:
Do we have to kill this function? Like how would a user render a surface-tree without the OutputRenderer now?
They would have to create the RenderElements and manually call draw on them, right?
So we could keep `draw_surface_tree` (and potentially `draw_layer_surface` or `draw_window`) as a thin wrapper around that to make custom drawing more viable.

Reply:
good idea, I will try to keep them as a wrapper

Reply:
I added a small wrapper to `draw_render_elements` and re-added `draw_surface_tree`, `draw_window` and `draw_layer_surface`